jablib/src/main/java/org/jabref/logic/ocr/OcrResult.java

```diff
@@ -0,0 +1,60 @@
+package org.jabref.logic.ocr;
+
+/**
+ * Represents the result of an OCR operation.
+ * Uses sealed classes to ensure type safety and avoid null parameters.
+ */
+public sealed interface OcrResult {
+
+    /**
+     * Represents a successful OCR operation with extracted text.
+     */
+    record Success(String text) implements OcrResult {
+        public Success {
+            // Convert null to empty string instead of throwing exception
```

Comment:
Why?

Do the callers pass null? We should avoid null in 2025

Annotate paramter with `@NonNull`